version: '3.8'

services:
  dynamo-sync:
    build: .
    environment:
      - AWS_DEFAULT_REGION=${AWS_DEFAULT_REGION:-ap-southeast-2}
      - AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
      - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
      - AWS_SESSION_TOKEN=${AWS_SESSION_TOKEN}
      - PGHOST=${PGHOST}
      - PGPORT=${PGPORT:-5432}
      - PGUSER=${PGUSER}
      - PGPASSWORD=${PGPASSWORD}
      - PGDATABASE=${PGDATABASE}
    volumes:
      - ./data:/app/data:Z
      - ./models:/app/models:Z
      - ./load:/app/load:Z
      - ./.env:/app/.env:Z
    restart: unless-stopped
    working_dir: /app
    command: python main.py --help
